packages/typespec-ts/src/modular/buildCodeModel.ts

```diff
@@ -1231,9 +1231,9 @@ function applyEncoding(
 
 function mergeFormatAndEncoding(
   format: string | undefined,
-  encoding: string,
+  encoding: string | undefined,
```

Comment:
Should have similar logic https://github.com/Azure/typespec-azure/pull/1258/files#diff-c390806a90d9e18d8cdd83b1f83c1bfca413262c0a0ad4a1c0576e3c6430d865